iroh/tests/provide.rs

```diff
@@ -417,56 +418,13 @@ async fn test_server_close() {
         .expect("supervisor failed");
 }
 
-#[cfg(feature = "flat-db")]
-#[tokio::test]
-async fn test_blob_reader_partial() -> Result<()> {
```

Comment:
I have no idea anymore what the test was supposed to test. I think it tested a behaviour of a complete download function - basically that it would behave correctly when abandoned in the middle.

But now that we have the finite state machine with explicit state transitions having such a test does not really make sense anymore.